app/services/iiif_print/manifest_builder_service_behavior.rb

```diff
@@ -71,27 +87,33 @@ def apply_v2_metadata_to_canvas(canvas:, presenter:)
 
     def sorted_canvases_v2(hash:, sort_field:)
       sort_field = Hyrax::Renderers::AttributeRenderer.new(sort_field, nil).label
-      hash["sequences"]&.first&.[]("canvases")&.sort_by! do |canvas|
-        selection = canvas["metadata"].select { |h| h["label"] == sort_field }
-        fallback = [{ label: sort_field, value: ['~'] }]
-        identifier_metadata = selection.presence || fallback
-        identifier_metadata.first["value"] if identifier_metadata.present?
+      hash['sequences']&.first&.[]('canvases')&.sort_by! do |canvas|
+        selection = canvas['metadata'].select { |h| h['label'] == sort_field }
+        fallback = [{ label: sort_field,
+                      value: ['~'] }]
+        sort_field_metadata = selection.presence || fallback
+        sort_field_metadata.first['value'] if sort_field_metadata.present?
       end
       hash
     end
 
-    def sorted_canvases_v3(hash:, **)
-      # TODO: flesh out metadata for v3
+    def sorted_canvases_v3(hash:, sort_field:)
+      sort_field = Hyrax::Renderers::AttributeRenderer.new(sort_field, nil).label
+      hash['items']&.sort_by! do |item|
+        selection = item['metadata'].select { |h| h['label'][I18n.locale.to_s] == [sort_field] }
+        fallback = [{ label: { "#{I18n.locale}": [sort_field] },
+                      value: { none: ['~'] } }]
+        sort_field_metadata = selection.presence || fallback
+        sort_field_metadata.first['value']['none'] if sort_field_metadata.present?
+      end
       hash
     end
 
-    def get_solr_docs(presenter)
+    def get_solr_hits(presenter)
       parent_id = presenter.id
       child_ids = presenter.try(:member_ids) || presenter.try(:ordered_ids)
-      parent_id_and_child_ids = child_ids << parent_id
-      query = ActiveFedora::SolrQueryBuilder.construct_query_for_ids(parent_id_and_child_ids)
-      solr_hits = ActiveFedora::SolrService.query(query, fq: "-has_model_ssim:FileSet", rows: 100_000)
-      solr_hits.flat_map { |solr_hit| ActiveFedora::SolrService.query("id:#{solr_hit.id}", rows: 100_000) }
+      ids = child_ids << parent_id
```

Comment:
A side-effect of `<<` is that it updates the object that you are "shoveling into" (which can create unexpected side-effects as later on the presenter's member_ids array will now have the parent_ids)